pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -0,0 +1,182 @@
+	core "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/builder"
+	fclient "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/defclient"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/parser"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/template"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ApplicationReconciler reconciles a Application object
+type applicationReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core.oam.dev,resources=applications,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core.oam.dev,resources=applications/status,verbs=get;update;patch
+
+// Reconcile process app event
+func (r *applicationReconciler) Reconcile(req ctrl.Request) (result ctrl.Result, gerr error) {
+
+	ctx := context.Background()
+	_log := r.Log.WithValues("application", req.NamespacedName)
+	app := new(v1alpha2.Application)
+	if err := r.Get(ctx, client.ObjectKey{
+		Name:      req.Name,
+		Namespace: req.Namespace,
+	}, app); err != nil {
+		if kerrors.IsNotFound(err) {
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if app.DeletionTimestamp != nil {
+		_log.Info("Handle delete")
+		owns := false
+
+		matchLabels := client.MatchingLabels{
+			builder.OamApplicationLable: app.Name,
+		}
+
+		aclist := &v1alpha2.ApplicationConfigurationList{}
+		if err := r.List(ctx, aclist, matchLabels); err != nil && !kerrors.IsNotFound(err) {
+			return ctrl.Result{}, errors.WithMessage(err, "list appConfigs")
+		}
+		for index := range aclist.Items {
+			ac := aclist.Items[index]
+			owns = true
+			if ac.DeletionTimestamp != nil {
+				continue
+			}
+
+			if err := r.Client.Delete(ctx, &ac); err != nil && !kerrors.IsNotFound(err) {
+				return ctrl.Result{}, errors.Errorf("delete ApplicationConfig: %s", ac.Name)
+			}
+		}
+
+		compList := &v1alpha2.ComponentList{}
+		if err := r.List(ctx, compList, client.MatchingLabels{
+			builder.OamApplicationLable: app.Name,
+		}); err != nil && !kerrors.IsNotFound(err) {
+			return ctrl.Result{}, errors.WithMessage(err, "list componetes")
+		}
+		for index := range compList.Items {
+			comp := compList.Items[index]
+			owns = true
+			if comp.DeletionTimestamp != nil {
+				continue
+			}
+			if err := r.Client.Delete(ctx, &comp); err != nil && !kerrors.IsNotFound(err) {
+				return ctrl.Result{}, errors.Errorf("delete Component: %s", comp.Name)
+			}
+		}
+		if !owns {
+			_log.Info("Remove finalizer")
+			removeFinalizers(app)
+			return ctrl.Result{}, r.Update(ctx, app)
+		}
+	}
+
+	if app.Status.Phase == "finished" {
+		return ctrl.Result{}, nil
+	}
```

Comment:
once the CR is reconciled succeed, it will never reconcile again?

Reply:
Synchronizing with appconfig and component may be a complicated process, and it is planned to be promoted through subsequent pr